### PR TITLE
Code-Switching dataset creation - upgrading to aggregate tokenizer manifest format

### PR DESCRIPTION
**Pull request by KunalDhawan**
# What does this PR do ?

Added functionality to create synthetic code-switching (CS) audio data and manifest in the multi-sample lid format. Added flag (--is_lid_manifest) to choose between standard flat text format and the multi-sample lid format,  to ensure backward compatibility. 

**Collection**:  ASR

# Changelog 
- in scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py,
    - added flag '--is_lid_manifest' which default to True
    - modified the 'create_cs_data' function to now create the text field in the format “text”: [{“str”:“esta muestra ” “lang”:”es”},{“str”:“was generated synthetically”: “lang”:”en”}]
    - updated the README to add the new functionality

# Usage
1. Create the (intermediate) manifest file using `code_switching_manifest_creation.py`.  It's usage is as follows:

    `python code_switching_manifest_creation.py --manifest_language1 <absolute path of Language 1's manifest file> --manifest_language2 <absolute path of Language 2's manifest file> --manifest_save_path <absolute path to save the created manifest> --id_language1 <language code for language 1 (e.g. en)> --id_language2 <language code for language 2 (e.g. es)> --max_sample_duration_sec <maximum duration of generated sample in seconds> --min_sample_duration_sec <maximum duration of generated sample in seconds> --dataset_size_required_hrs <size of generated synthetic dataset required in hrs>`

2. Create the synthetic audio data and the corresponding manifest file using `code_switching_audio_data_creation.py` It's usage is as follows: 

    `python code_switching_audio_data_creation.py --manifest_path <absolute path to intermediate CS manifest generated in step 1> --audio_save_folder_path <absolute path to directory where you want to save the synthesized audios> --manifest_save_path <absolute path to save the created manifest> --audio_normalized_amplitude <scaled normalized amplitude desired> --cs_data_sampling_rate <desired sampling rate for generated audios> --sample_beginning_pause_msec <pause to be added to the beginning of the generated sample in milli seconds> --sample_joining_pause_msec <pause to be added between segments while joining, in milli seconds> --sample_end_pause_msec <pause to be added to the end of the generated sample in milli seconds> --is_lid_manifest <boolean to create manifest in the multi-sample lid format for the text field, true by default> --workers <number of worker processes>`

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

File: scripts/speech_recognition/code_switching/README.md
Comment (bmwshop) [inline scripts/speech_recognition/code_switching/README.md:1]: maybe call this parameter `--is_lid_manifest` ? It's not directly related to aggregate tokenizer[s] per se

File: scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py
Comment (bmwshop) [inline scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py:1]: same comment as above

File: scripts/speech_recognition/code_switching/README.md
Comment (KunalDhawan) [reply scripts/speech_recognition/code_switching/README.md:1]: updated the variable name

File: scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py
Comment (KunalDhawan) [reply scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py:1]: addressed in latest commit

**Review by bmwshop — APPROVED** [review]

**Pull request merged**